pincer/commands/commands.py

```diff
@@ -168,169 +183,337 @@ async def test_command(
+        :class:`~objects.message.context.MessageContext`,
+        :class:`~objects.user.user.User`,
+        :class:`~objects.guild.member.GuildMember`,
 
-        param_type = _options_type_link.get(annotation)
 
-        if not param_type:
-            raise InvalidArgumentAnnotation(
-                f"Annotation `{annotation}` on parameter "
-                f"`{param}` in command `{cmd}` (`{func.__name__}`) is not "
-                "a valid type."
-            )
+    Parameters
+    ----------
+    name : Optional[:class:`str`]
+        The name of the command |default| :data:`None`
+    enable_default : Optional[:class:`bool`]
+        Whether the command is enabled by default |default| :data:`True`
+    guild : Optional[Union[:class:`~pincer.utils.snowflake.Snowflake`, :class:`int`, :class:`str`]]
+        What guild to add it to (don't specify for global) |default| :data:`None`
+    cooldown : Optional[:class:`int`]
+        The amount of times in the cooldown_scale the command can be invoked
+        |default| ``0``
+    cooldown_scale : Optional[:class:`float`]
+        The 'checking time' of the cooldown |default| ``60``
+    cooldown_scope : :class:`~pincer.objects.app.throttle_scope.ThrottleScope`
+        What type of cooldown strategy to use |default| :attr:`ThrottleScope.USER`
 
-        options.append(
-            AppCommandOption(
-                type=param_type,
-                name=param,
-                description=argument_description or "Description not set",
-                required=required,
-                choices=choices or MISSING,
-            )
+    Raises
+    ------
+    CommandIsNotCoroutine
+        If the command function is not a coro
+    InvalidCommandName
+        If the command name does not follow the regex ``^[\\w-]{1,32}$``
+    InvalidCommandGuild
+        If the guild id is invalid
+    CommandDescriptionTooLong
+        Descriptions max 100 characters
+        If the annotation on an argument is too long (also max 100)
+    CommandAlreadyRegistered
+        If the command already exists
+    InvalidArgumentAnnotation
+        Annotation amount is max 25,
+        Not a valid argument type,
+        Annotations must consist of name and value
+    """
+    # noqa: E501
+    return register_command(
+        func=func,
+        app_command_type=AppCommandType.USER,
+        name=name,
+        enable_default=enable_default,
+        guild=guild,
+        cooldown=cooldown,
+        cooldown_scale=cooldown_scale,
+        cooldown_scope=cooldown_scope
+    )
+
+
+def message_command(
+    func=None,
+    *,
+    name: Optional[str] = None,
+    enable_default: Optional[bool] = True,
+    guild: Union[Snowflake, int, str] = None,
+    cooldown: Optional[int] = 0,
+    cooldown_scale: Optional[float] = 60,
+    cooldown_scope: Optional[ThrottleScope] = ThrottleScope.USER,
+):
+    """A decorator to create a user command to register and respond to
+    with the discord API from a function.
```

Comment:
Would rather do this, unless I'm misunderstanding the use of this decorator (sorry if that's the case)
```suggestion
    """A decorator to make a user command register and respond
    with the Discord API from a function.
```